test/common/pool.c

```diff
@@ -0,0 +1,194 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "pool.h"
+
+#include "provider.h"
+#include <umf/memory_pool_ops.h>
+
+#include <assert.h>
+#include <stdlib.h>
+
+static enum umf_result_t nullInitialize(umf_memory_provider_handle_t *providers,
```

Comment:
IMO there should be "providers" and "pools" dirs in main structure. The NULL and tracking providers/pools should be put into separate folders there

Reply:
Done.